### PR TITLE
Vertex AI Gemini: support system messages and parallel tools/functions

### PR DESCRIPTION
**Pull request by glaforge**
- support system instructions
- support parallel function calling
- should avoid the allocation issue faced sometimes because of the lifecycle of the underlying VertexAI client object

### CODE REVIEW

File: .idea/icon.png
Comment (langchain4j) [inline .idea/icon.png:1]: :cry:

File: langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/VertexAiGeminiChatModelIT.java [hunk 1]
```diff
@@ -35,16 +35,19 @@ class VertexAiGeminiChatModelIT {
     static final String CAT_IMAGE_URL = "https://upload.wikimedia.org/wikipedia/commons/e/e9/Felis_silvestris_silvestris_small_gradual_decrease_of_quality.png";
     static final String DICE_IMAGE_URL = "https://upload.wikimedia.org/wikipedia/commons/4/47/PNG_transparency_demonstration_1.png";
 
+    public static final String GEMINI_1_0_PRO_VISION = "gemini-1.5-pro-preview-0514";
+    public static final String GEMINI_PRO = "gemini-1.5-pro-preview-0514";
```
Comment (langchain4j) [inline langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/VertexAiGeminiChatModelIT.java]: nit:
```suggestion
    public static final String GEMINI_1_5_PRO = "gemini-1.5-pro-preview-0514";
```

File: langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/VertexAiGeminiStreamingChatModelIT.java [hunk 2]
```diff
@@ -34,16 +34,19 @@
 
 class VertexAiGeminiStreamingChatModelIT {
 
+    public static final String GEMINI_PRO = "gemini-1.5-pro-preview-0514";
+    public static final String GEMINI_PRO_VISION = "gemini-1.5-pro-preview-0514";
```
Comment (langchain4j) [inline langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/VertexAiGeminiStreamingChatModelIT.java]: is it intended that both constants refer to the same (gemini-1.5-pro-preview-0514) model?

**Review by ghost — COMMENTED** [review]
@glaforge looks great, thank you!

File: langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/VertexAiGeminiStreamingChatModelIT.java (hunk 2 above)
Comment (glaforge) [reply langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/VertexAiGeminiStreamingChatModelIT.java]: Good point, there's only really one model now, so there's no distinction between multimodal or not.
I'll update to just use one constant.

File: langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/VertexAiGeminiChatModelIT.java (hunk 1 above)
Comment (glaforge) [reply langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/VertexAiGeminiChatModelIT.java]: Will use `GEMINI_1_5_PRO` as suggested.

File: langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/VertexAiGeminiStreamingChatModelIT.java (hunk 2 above)
Comment (glaforge) [reply langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/VertexAiGeminiStreamingChatModelIT.java]: Now using just one constant / single model.

**Review by ghost — APPROVED** [review]
@glaforge great job, thank you!

**Pull request merged**